### PR TITLE
Allow re-running the plugin script

### PR DESCRIPTION
**Pull request by sridhargaddam**
In an existing KIND deployment when certain make commands are re-run, the external-net plugin script is returning error as the required resources already exist. This PR modifies the script to check if the resource already exists and skip that step.

Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

**Review by skitt — APPROVED** [review]

File: scripts/e2e/external/utils [hunk 1]
```diff
@@ -17,6 +17,17 @@ function get_ip_on_container_network() {
     docker inspect "${container_name}" -f '{{(index .NetworkSettings.Networks "'"${network_name}"'").IPAddress}}'
 }
 
+function docker_network_connect() {
+    network_name=$1
+    container_name=$2
```
Comment (mkolesnik) [inline scripts/e2e/external/utils]:
```suggestion
    local container=$1
```

Use `local` for function local variables.
Also no need to pass the network name as it's always the same.

File: scripts/e2e/external/utils [hunk 2]
```diff
@@ -48,24 +80,23 @@ function setup_external() {
...
+                docker_network_connect "${EXTERNAL_NET}" "${extcluster}-${node}${num_worker}"
             fi
             num_worker=$(( num_worker + 1))
         fi
     done
 
     # Create external test container on pseudo-ext network
-    docker run --cap-add=NET_ADMIN --name "${EXTERNAL_APP}" --network "${EXTERNAL_NET}" \
-        -d registry.access.redhat.com/ubi7/ubi python -m SimpleHTTPServer 1234
+    docker_create_ext_container ${EXTERNAL_APP} ${EXTERNAL_NET}
```
Comment (mkolesnik) [inline scripts/e2e/external/utils]:
```suggestion
    docker inspect --type=container "${EXTERNAL_APP}" > /dev/null 2>&1 || \
        docker run --cap-add=NET_ADMIN --name "$EXTERNAL_APP" --network "$EXTERNAL_NET" \
            -d registry.access.redhat.com/ubi7/ubi python -m SimpleHTTPServer 1234
```

Or more simply (if you add `docker_exists`):

```suggestion
    docker_exists container "${EXTERNAL_APP}" || \
        docker run --cap-add=NET_ADMIN --name "$EXTERNAL_APP" --network "$EXTERNAL_NET" \
            -d registry.access.redhat.com/ubi7/ubi python -m SimpleHTTPServer 1234
```

File: scripts/e2e/external/utils [hunk 3]
```diff
@@ -32,9 +43,30 @@ function delete_container_network() {
...
+            -d registry.access.redhat.com/ubi7/ubi python -m SimpleHTTPServer 1234
+    else
+        echo "Docker container ${external_app} already exists, skipping its creation..."
+    fi
+}
+
 function setup_external() {
     # Create pseudo-ext docker network
-    docker network create --driver=bridge --subnet="${EXTERNAL_CIDRS}" "${EXTERNAL_NET}"
+    create_container_network "${EXTERNAL_NET}" ${EXTERNAL_CIDRS}
```
Comment (mkolesnik) [inline scripts/e2e/external/utils]:
```suggestion
    docker inspect --type=network "$EXTERNAL_NET" > /dev/null 2>&1 || \
        docker network create --driver=bridge --subnet="$EXTERNAL_CIDRS" "$EXTERNAL_NET"
```

Or more simply (if you add `docker_exists`):

```suggestion
    docker_exists network "$EXTERNAL_NET" || \
        docker network create --driver=bridge --subnet="$EXTERNAL_CIDRS" "$EXTERNAL_NET"
```

File: scripts/e2e/external/utils [hunk 4]
```diff
@@ -17,6 +17,17 @@ function get_ip_on_container_network() {
     docker inspect "${container_name}" -f '{{(index .NetworkSettings.Networks "'"${network_name}"'").IPAddress}}'
 }
```
Comment (mkolesnik) [inline scripts/e2e/external/utils:19]:
```suggestion

function docker_exists() {
    local type=$1
    local name=$2
    local format=$3
    docker inspect --type="$type" -f "$format" "$name" > /dev/null 2>&1
}
```

Then you can use it everywhere you want to test if something is in docker.
And yes, omitting the 3rd argument works just fine, e.g. `docker_exists network "$EXTERNAL_NET"` works as you expect.

File: scripts/e2e/external/utils [hunk 5]
```diff
@@ -17,6 +17,17 @@ function get_ip_on_container_network() {
...
 
+function docker_network_connect() {
+    network_name=$1
+    container_name=$2
+    if docker inspect "${container_name}" -f '{{(index .NetworkSettings.Networks "'"${network_name}"'").IPAddress}}' >/dev/null 2>&1; then
+      echo "Docker container ${container_name} already has an interface on network ${network_name}"
+      return
+    fi
+
+    docker network connect "${network_name}" "${container_name}"
```
Comment (mkolesnik) [inline scripts/e2e/external/utils]:
```suggestion
    docker inspect "$container" -f "{{(index .NetworkSettings.Networks \"${EXTERNAL_NET}\").IPAddress}}" >/dev/null 2>&1 || \
        docker network connect "$EXTERNAL_NET" "$container"
```

Or more simply (if you add `docker_exists`):
```suggestion
    docker_exists container "$container" "{{(index .NetworkSettings.Networks \"${EXTERNAL_NET}\").IPAddress}}" || \
        docker network connect "$EXTERNAL_NET" "$container"
```

**Review by mkolesnik — CHANGES_REQUESTED** [review]

File: scripts/e2e/external/utils (hunk 1 above)
Comment (sridhargaddam) [reply scripts/e2e/external/utils]: Thanks for reviewing @mkolesnik

File: scripts/e2e/external/utils [hunk 6]
```diff
@@ -32,9 +46,31 @@ function delete_container_network() {
...
+    external_app=$1
+    external_net=$2
+    if [[ $(docker ps --filter name="${external_app}" --format="{{ .Names }}" | wc -l) -eq 0 ]]; then
+        docker run --cap-add=NET_ADMIN --name "${external_app}" --network "${external_net}" \
+            -d registry.access.redhat.com/ubi7/ubi python -m SimpleHTTPServer 1234
+    else
+        echo "Docker container ${external_app} already exists, skipping its creation..."
+    fi
+}
+
```
Comment (mkolesnik) [inline scripts/e2e/external/utils]:
```suggestion
```

These aren't needed anymore :)

**Review by mkolesnik — CHANGES_REQUESTED** [review]

File: scripts/e2e/external/utils (hunk 6 above)
Comment (sridhargaddam) [reply scripts/e2e/external/utils]: Oh yeah, missed removing them.

**Review by mkolesnik — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Pull request merged**